implementations/rust/ockam/ockam_api/src/nodes/service/tcp_inlets/node_manager.rs

```diff
@@ -109,135 +118,162 @@ impl NodeManager {
-            let result = session
-                .initial_connect()
-                .await
-                .map(|outcome| match outcome {
-                    ReplacerOutputKind::Inlet(status) => status,
-                    _ => {
-                        panic!("Unexpected outcome: {:?}", outcome)
-                    }
-                });
-
-            match result {
-                Ok(status) => Some(status),
-                Err(err) => {
-                    warn!("Failed to create inlet: {err}");
-                    None
-                }
+        if wait_connection {
+            let result = first_session.initial_connect().await;
+            if let Err(error) = result {
+                warn!(%error, "Failed to connect to the outlet");
             }
-        } else {
-            None
         };
 
-        let connection_status = session.connection_status();
+        first_session.start_monitoring()?;
+        sessions.push(first_session);
 
-        session.start_monitoring()?;
+        let inlet_info = TcpInletHandle::new(inlet, outlet_addresses.clone(), sessions, privileged);
 
-        self.registry.inlets.insert(
-            alias.clone(),
-            InletInfo::new(
-                &listen_addr.to_string(),
-                outlet_address.clone(),
-                session,
-                privileged,
-            ),
-        );
-
-        let tcp_inlet_status = InletStatus::new(
-            listen_addr.to_string(),
-            outcome
-                .clone()
-                .and_then(|s| s.worker.map(|address| address.address().to_string())),
-            &alias,
-            None,
-            outcome.clone().map(|s| s.route.to_string()),
-            connection_status,
-            outlet_address.to_string(),
-            privileged,
-        );
+        // this summary already contains the connection status
+        let summary = inlet_info.summary().await;
+
+        self.registry.inlets.insert(alias.clone(), inlet_info);
+
+        let tcp_inlet_status =
+            InletStatusView::new(listen_addr, &alias, &outlet_addresses, privileged, summary);
 
         info! {
             %listen_address,
-            %outlet_address,
+            outlet_addresses = %outlet_addresses_str,
             %alias,
             "inlet created"
         }
 
         Ok(tcp_inlet_status)
     }
 
-    pub async fn delete_inlet(&self, alias: &str) -> Result<InletStatus> {
+    pub async fn delete_inlet(&self, context: &Context, alias: &str) -> Result<InletStatusView> {
         info!(%alias, "Handling request to delete inlet portal");
         if let Some(inlet_to_delete) = self.registry.inlets.remove(alias) {
             debug!(%alias, "Successfully removed inlet from node registry");
-            inlet_to_delete.session.lock().await.stop().await;
+            if let Err(error) = inlet_to_delete.stop(context).await {
+                error!(%alias, %error, "Failed to stop inlet");
```

Comment:
If we just log the error, should it be a warning instead?

Reply:
My reasoning is that it should never happen (likely a bug), and if we fail to stop an inlet, the relative port will not available anymore for other usages.